app/src/main/java/com/example/scoutconcordia/CalendarActivity.java

```diff
@@ -0,0 +1,226 @@
+                    @Override
+                    public void onClick(DialogInterface dialog, int id) {
+
+                        displayCalendar(calendarIds.get(selectedCalendar));
+                    }
+                })
+                .setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int id) {
+                        Intent mapIntent = new Intent(CalendarActivity.this, MapsActivity.class);
+                        startActivity(mapIntent);
+                        CalendarActivity.this.overridePendingTransition(0, 0);
+                    }
+                });
+
+        return builder.create();
+    }
+
+    public void showCalendarsList(Calendar service){
+        HashMap<String, String> calendarIds = new HashMap<String, String>();
+        ArrayList<String> calendars =  new ArrayList<>();
+        try {
+            String pageToken = null;
+            do {
+                CalendarList calendarList = service.calendarList().list().setPageToken(pageToken).execute();
+                List<CalendarListEntry> items = calendarList.getItems();
+
+                for (CalendarListEntry calendarListEntry : items) {
+                    //Log.d("CALENDARSSS: ",calendarListEntry.getSummary());
+                    calendars.add(calendarListEntry.getSummary());
+                    calendarIds.put(calendarListEntry.getSummary(), calendarListEntry.getId());
+                }
+                pageToken = calendarList.getNextPageToken();
+            } while (pageToken != null);
+
+//            return
+        }
+        catch(Throwable t){
+        }
+        Dialog dialog = onCreateDialogSingleChoice(calendars, calendarIds);
+        dialog.show();
+    }
+
+    protected void signIn(){
+        Intent signInIntent = googleSignInClient.getSignInIntent();
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+
+    // result of the starActivityForResult in signIn()
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data){
+        super.onActivityResult(requestCode, resultCode, data);
+
+        // Verifies if the requqest code is that of the startActivityForResult using signInIntent in signIn()
+        if(requestCode == RC_SIGN_IN){
+            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+            handleSignInresult(task);
+        }
+    }
+
+    // handles the behavious once the signning in has completed
+    protected void handleSignInresult(Task<GoogleSignInAccount> task){
+        try{
+            GoogleSignInAccount account = task.getResult(ApiException.class);
+            email = account.getEmail();
+            new RetrieveAccessToken().execute();
+            Toast.makeText(this, email, Toast.LENGTH_LONG).show();
+        }catch(ApiException e){
+            Log.d(TAG, e.getMessage());
+        }
+    }
+
+    @Override
```

Comment:
Why is there an override annotation for onPause() when the method is simply calling super.onPause()?